python/pyarrow/type_traits.pxi

```diff
@@ -0,0 +1,394 @@
+                                        is_dictionary,
+                                        is_fixed_size_binary,
+                                        is_fixed_width,
+                                        is_var_length_list,
+                                        is_list,
+                                        is_list_like,
+                                        is_var_length_list_like,
+                                        is_list_view,
+                                        is_nested,
+                                        is_union,
+                                        bit_width,
+                                        offset_bit_width)
+
+
+def is_integer_type(data_type):
+    """
+    Check if the data type is an integer type.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+    """
+    return is_integer(data_type.id)
+
+
+def is_signed_integer_type(data_type):
+    """
+    Check if the data type is a signed integer type.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+    """
+    return is_signed_integer(data_type.id)
+
+
+def is_unsigned_integer_type(data_type):
+    """
+    Check if the data type is an unsigned integer type.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+    """
+    return is_unsigned_integer(data_type.id)
+
+
+def is_floating_type(data_type):
+    """
+    Check if the data type is a floating type.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+    """
+    return is_floating(data_type.id)
+
+
+def is_numeric_type(data_type):
+    """
+    Check if the data type is a numeric type.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+    """
+    return is_numeric(data_type.id)
+
+
+def is_decimal_type(data_type):
+    """
+    Check if the data type is a decimal type.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+    """
+    return is_decimal(data_type.id)
+
+
+def is_run_end_type_py(data_type):
+    """
+    Check if the data type is a run end type.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+    """
+    return is_run_end_type(data_type.id)
+
+
+def is_primitive_type(data_type):
+    """
+    Check if the data type is a primitive type.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+    """
+    return is_primitive(data_type.id)
+
+
+def is_base_binary_like_type(data_type):
+    """
+    Check if the data type is a base_binary_like type.
+
+    Parameters
+    ----------
+    data_type : DataType
+        The data type to check
+    """
+    return is_base_binary_like(data_type.id)
+
+
+def is_binary_like_type(data_type):
+    """
+    Check if the data type is a _binary_like type.
```

Comment:
These docstrings are not useful if they merely repeat what the function name says. You should explain what a "_binary_like" type is, because the reader probably doesn't know.

Reply:
I have enhanced the content.

Reply:
It is looking much better, thanks! I think you could remove the first sentence, and maybe the last ("Useful for ..."). The middle portions of the docstrings are quite clear already.